src/main/java/com/fulcrumgenomics/util/PickLongIndices.scala

```diff
@@ -0,0 +1,291 @@
+  type Index = Array[Byte]
+}
+
+/** A set of indices that maintains things as both Strings and byte[]s. */
+private class IndexSet extends Iterable[Index] {
+  private val arrays = ListBuffer[Array[Byte]]()
+  private val set    = mutable.HashSet[String]()
+
+  override def iterator: Iterator[Array[Byte]] = this.arrays.iterator
+
+  /** Add an index to the set. */
+  def +=(index : Array[Byte]): Unit = {
+    this.arrays += index
+    this.set.add(new String(index))
+  }
+
+  /** Checks to see whether an index is contained in the set. */
+  def contains(index: Array[Byte]): Boolean = contains(new String(index))
+
+  /** Checks to see whether an index is contained in the set. */
+  def contains(index: String): Boolean = this.set.contains(index)
+}
+
+@clp(group=ClpGroups.Utilities, description =
+    """
+      |Picks a set of molecular indices that have at least a given number
+      |of mismatches between them.
+    """)
+class PickLongIndices
+(
+  @arg(flag="l", doc="The length of each barcode sequence.")                           val length: Int = 8,
+  @arg(flag="n", doc="The number of indices desired.")                                 val numberOfIndices: Int,
+  @arg(flag="e", doc="The minimum edit distance between two indices in the set.")      val editDistance: Int = 3,
+  @arg(flag="o", doc="File to write indices to.")                                      val output: FilePath,
+  @arg(          doc="Allow indices that are lexical reverses of one another")         val allowReverses: Boolean = false,
+  @arg(          doc="Allow indices that are reverse complements of one another")      val allowReverseComplements: Boolean = false,
+  @arg(          doc="Allow indices that are palindromic (index == revcomp(index)).")  val allowPalindromes: Boolean = false,
+  @arg(          doc="Reject indices with a homopolymer of greater than this length.") val maxHomopolymer: Int = 2,
+  @arg(flag="g", doc="The minimum GC fraction for a barcode to be accepted.")          val minGc: Double = 0.2,
+  @arg(flag="G", doc="The maximum GC fraction for a barcode to be accepted.")          val maxGc: Double = 0.8,
+  @arg(          doc="File of existing index sequences to integrate, one per line.")   val existing: Option[FilePath] = None,
+  @arg(flag="s", doc="Random seed value.")                                             val seed: Int = 1,
+  @arg(flag="a", doc="Attempts to pick the next index before quitting.")               val attempts: Int = 1e6.toInt,
+  @arg(          doc="The installation directory for ViennaRNA.")                      val viennaRnaDir: Option[DirPath] = None,
+  @arg(flag="t", doc="The temperature at which to predict secondary structure.")       val temperature: Double = 25d,
+  @arg(          doc="The lowest acceptable secondary structure deltaG.")              val minDeltaG: Double = -10,
+  @arg(          doc="Adapter sequence(s) indices will be inserted into.", minElements=0) val adapters: Seq[String] = Seq(),
```

Comment:
Adapter sequence(s) into which indices will be inserted?  (end with a preposition?)